### PR TITLE
ContextMenu: fixes add annotation button to work on old graph

### LINKED ISSUE #31911: Add Annotation button does not work on old graph panel
**What happened**:
Clicking the annotation button in the old graph does not prompt me to add any data.

https://user-images.githubusercontent.com/29557702/110815973-379c7e80-828b-11eb-9432-4e5bb14f6b2d.mov

**What you expected to happen**:
It should show the add annotation form.
**How to reproduce it (as minimally and precisely as possible)**:
Go to dashboard.
Create a panel.
Click on the graph panel to add an annotation.
Click the "add annotation button"

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR fixes add annotation button to work on the old graph. Also added support for keyboard events and onClose handler.
**Which issue(s) this PR fixes**:

Fixes #31911

### CODE REVIEW

File: packages/grafana-ui/src/components/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -24,9 +24,20 @@ export interface MenuItemProps {
   /** Active */
   active?: boolean;
 }
+/** This interface was created because we are utilizing two different events in the onClick props*/
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Menu/MenuItem.tsx:27]: Not sure. understand why here

File: packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx [hunk 2]
```diff
@@ -73,7 +73,23 @@ export const ContextMenu: React.FC<ContextMenuProps> = React.memo(({ x, y, onClo
...
+                  // We can have both url and onClick and we want to allow user to open the link in new tab/window
+                  const isSpecialKeyPressed = e.ctrlKey || e.metaKey || e.shiftKey;
+                  if (isSpecialKeyPressed && item.url) {
+                    return;
+                  }
+
+                  if (item.onClick) {
+                    e.preventDefault();
+                    item.onClick(e);
+                  }
```
Comment (torkelo) [inline packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:86]: this was not needed before, not sure we need this logic now. 

Can you test 
https://github.com/grafana/grafana/pull/31915

File: packages/grafana-ui/src/components/Menu/MenuItem.tsx (hunk 1 above)
Comment (thisisobate) [reply packages/grafana-ui/src/components/Menu/MenuItem.tsx:27]: So the onClick handler for ContextMenu is coming from two different sources: menuItem data object and onclose (ContextMenu Props).
We need a way to compose these two actions and that's why we had this conditional logic in the contextMenu component:
```
onClick={(e: React.MouseEvent<HTMLElement>) => {
                  // We can have both url and onClick and we want to allow user to open the link in new tab/window
                  const isSpecialKeyPressed = e.ctrlKey || e.metaKey || e.shiftKey;
                  if (isSpecialKeyPressed && item.url) {
                    return;
                  }

                  if (item.onClick) {
                    e.preventDefault();
                    item.onClick(e);
                  }
```

Now, the challenge is that onclick prop was set to type SynthenticEvent which doesn't correspond with the MouseEvent type we're using here, thus, it requires we create a new interface just to allow onclick accept MouseEvent actions. @torkelo

File: packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx (hunk 2 above)
Comment (thisisobate) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:86]: ok...doing that now

Comment (thisisobate) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:86]: Works as expected

Comment (thisisobate) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:86]: > this was not needed before, not sure we need this logic now.
Actually, this was the initial logic we had prior to the menu component refactor.

Comment (thisisobate) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:86]: Do we remove this logic? @torkelo

Comment (torkelo) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:86]: Can you link to where this was done before ?

Comment (thisisobate) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:86]: Here you go: https://github.com/grafana/grafana/blob/a3d765830e7f4783c58e7a9d8713e1289fdbc3ea/packages/grafana-ui/src/components/Menu/Menu.tsx#L93

Comment (torkelo) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:86]: Yes, but that is part of the MenuItem component, why should that be added to ContextMenu? 

But even then I think that code is not needed, tested and open links in new tab works great

Comment (thisisobate) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:86]: Alright 👍

**Comment by thisisobate** [comment]
Fixed! #31915

**Pull request closed without merging**